imx/signers/stark/factory.go

```diff
@@ -11,13 +11,18 @@ import (
 )
 
 // package-global variable to represent our standard stark curve
-// generally bad practice, but has no state + we never configure it
-// unexported so can be updated in future if necessary
-// TODO: should we do this in init() to allow us to handle the eaten error here
```

Comment:
I came to see why GenerateKey needs to return error. 
I think the suggestion in this TODO is worth exploring. 
Have you explored it and found a problem therefore removing this TODO comment?

Reply:
Yes, the solution is implemented in the following lines.